app/src/main/java/com/android/sample/model/userAccount/UserAccountViewModel.kt

```diff
@@ -151,6 +152,7 @@ open class UserAccountViewModel(
   }
 
   fun acceptFriendRequest(friendId: String) {
+    updateReceivedListAfterRemoval(friendId)
```

Comment:
Wouldn't it be more logical to add this updateReceivedListAfterRemoval() method only after successfully accepting the friend request so that the list isn't updating if there's an issue ?

Reply:
The reason why I do this is because I want to create the illusion of a responsive UI. I want to manually update the local list before waiting for the response of the cloud services. In the very unlikely event that the operation will fail, on the next composition of the composable i will fetch the correct backend data:
`LaunchedEffect(Unit) { userAccountViewModel.fetchReceivedRequests() }`
The concept is called optimistic updates. Here is an article explaining the idea: [What are optimistic updates?](https://medium.com/@kyledeguzmanx/what-are-optimistic-updates-483662c3e171)

Reply:
Great idea, I thought you weren't handling optimistic update failures properly but I was wrong. Nice job !